src/service.rs

```diff
@@ -524,13 +523,18 @@ impl Service {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn start<S: AsRef<OsStr>>(&self, service_arguments: &[S]) -> Result<()> {
+    pub fn start<S: AsRef<OsStr>>(&self, service_arguments: &[S]) -> Result<(), Error> {
         let wide_service_arguments = service_arguments
             .iter()
-            .map(|s| WideCString::from_str(s).chain_err(|| ErrorKind::InvalidStartArgument))
-            .collect::<Result<Vec<WideCString>>>()?;
-        let mut raw_service_arguments: Vec<*const u16> =
-            wide_service_arguments.iter().map(|s| s.as_ptr()).collect();
+            .map(|s| WideCString::from_str(s).map_err(|_| ErrorKind::InvalidStartArgument))
+            .collect::<Vec<Result<WideCString, _>>>()
+            .into_iter()
+            .collect::<Result<Vec<WideCString>, _>>()?;
```

Comment:
Do you really need to `collect` twice? I think the magic of how `collect` works should be able to transform it into a `Result<Vec<...>, _>` directly?

Reply:
yeah that was unnecessary, my fault for not getting rid of it earlier.